### ISSUE TITLE
the "RES" of  microros agent keep increasing

### ISSUE THREAD

**Issue by chrisjaxy** [micro-ROS/micro_ros_setup#437]
## Issue template

- Hardware description: STM32F103
- RTOS: freertos
- Installation type: micro_ros_setup
- Version or commit hash: foxy

#### Steps to reproduce the issue
we created a publisher and a subscription on STM32F103 and created a test.sh regularly published irrelevant topic. I found micro_ros_agent RES keep increasing.
When we created 7 publishers, 4 subscribers, 10 service servers. I found the RES Increase faster. What could be the cause? Can this problem be solved?
test.sh
```
#!/bin/bash
while true
do 
   ros2 topic pub -t 1 /vel_mcu geometry_msgs/msg/Twist
done
```

a publisher and a subscription
<img width="637" alt="1" src="https://user-images.githubusercontent.com/95396234/146521534-101bc179-1dd1-4bca-a6ac-b3d098139e08.png">
<img width="630" alt="2" src="https://user-images.githubusercontent.com/95396234/146521541-9e1ff8f3-94a5-4da2-9283-4eea6ba50e53.png">
<img width="633" alt="3" src="https://user-images.githubusercontent.com/95396234/146521571-c369bb5c-aeab-4a8b-a8b3-afc767c350a6.png">

6 publishers, 4 subscribers, 10 service servers
<img width="630" alt="4" src="https://user-images.githubusercontent.com/95396234/146521642-5c3c7c2c-70ca-4caa-ab75-a9bd03cac03a.png">
<img width="634" alt="5" src="https://user-images.githubusercontent.com/95396234/146521652-e03782f3-2938-4748-adae-4f7fb69f2e29.png">

**Comment by pablogs9** [comment]
Which are the QoS of those micro-ROS entities?

**Comment by chrisjaxy** [comment]
> Which are the QoS of those micro-ROS entities?

- a publisher and a subscription
a default publisher, a best_effort subscription.
- 6 publishers, 4 subscribers, 10 service servers
3 default publisher, 3 best_effort publisher, 3 best_effort subscriptions, 1 default publisher, 10 best service servers.

"/vel_mcu" is an unrelated topic.

**Comment by pablogs9** [comment]
We will try to replicate this a find a solution.

**Comment by pablogs9** [comment]
Solved here: https://github.com/micro-ROS/micro-ROS-Agent/pull/107

**Comment by chrisjaxy** [comment]
> Solved here: [micro-ROS/micro-ROS-Agent#107](https://github.com/micro-ROS/micro-ROS-Agent/pull/107)
> 
> @chrisjaxy can you test this branch with your use case in order to ensure that it is solved?

Thank you for your help, we will test on our machine and tell you the conclusion.

**Comment by chrisjaxy** [comment]
> Solved here: [micro-ROS/micro-ROS-Agent#107](https://github.com/micro-ROS/micro-ROS-Agent/pull/107)
> 
> @chrisjaxy can you test this branch with your use case in order to ensure that it is solved?

Thank you for your help, I think it has solved the issue

**Comment by pablogs9** [comment]
Nice, I'm closing.
